APILambda/user/getUser.ts

```diff
@@ -51,25 +51,61 @@ const toUserWithRelationResponse = (user: DatabaseUserWithRelation) => ({
   }
 })
 
-const userAndRelationsWithId = (
+const getProfileNameWithBlockStatus = (
   conn: SQLExecutable,
   userId: string,
   fromUserId: string
 ) => {
   return conn.queryFirstResult<DatabaseUserWithRelation>(
     `
-    SELECT *, 
-    ur1.status AS themToYouStatus, 
-    ur2.status AS youToThemStatus 
-    FROM user u 
+    SELECT u.name, 
+    ur1.status AS themToYouStatus,
+    ur2.status AS youToThemStatus
+    FROM user u
     LEFT JOIN userRelations ur1 ON ur1.fromUserId = u.id
     AND ur1.fromUserId = :userId
     AND ur1.toUserId = :fromUserId
     LEFT JOIN userRelations ur2 ON ur2.toUserId = u.id
     AND ur2.fromUserId = :fromUserId
     AND ur2.toUserId = :userId
-    WHERE u.id = :userId;
+    WHERE u.id = :userId
+    AND ur1.status = 'blocked';
   `,
     { userId, fromUserId }
   )
 }
+
+const userAndRelationsWithId = (
+  conn: SQLExecutable,
+  userId: string,
+  fromUserId: string
+) =>
+  getProfileNameWithBlockStatus(conn, userId, fromUserId)
+    .flatMapSuccess((result) => {
+      if (
+        result &&
+        Object.keys(result).length !== 0 &&
+        result.themToYouStatus === "blocked"
+      ) {
+        return promiseResult(success(result))
+      }
+      return promiseResult(failure("no result found" as const))
+    })
+    .flatMapFailure(() => {
+      return conn.queryFirstResult<DatabaseUserWithRelation>(
+        `
+      SELECT *, 
```

Comment:
may be a way to do this without duplicating the query, but we can update that later